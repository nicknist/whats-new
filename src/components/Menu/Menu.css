body {
  color: black;
}

.yellow {
  color: yellow;
}

label {
  font-size: 1.5em;
  padding: 5%;
  display: flex;
}

.item:hover {
  background-color: #55A5FF;
  color: black;
  cursor: pointer;
}

.item:active {
  border: 2px solid blue;
}

.icon {
  height: 25px;
  width: 25px;
  display: inline;
  margin-right: 10px;
  background-color: white;
  border-radius: 50%;
}
