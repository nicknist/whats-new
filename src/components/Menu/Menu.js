import React from 'react';
import './Menu.css'
import location from '../../icons/location.svg'
import brain from '../../icons/brain.png'
import movie from '../../icons/movie.png'
import rocket from '../../icons/rocket.png'
import heart from '../../icons/heart.png'
import PropTypes from 'prop-types'

// MENU COMPONENT CODE GOES HERE

const Menu = ({ changePage }) => {
  return (
    <nav className="menu">
      <label onClick={changePage} id="local" className="item">
        <img className="icon" src={location} alt="Location Icon"/>
        Local News
      </label>
      <label onClick={changePage} id="technology" className="item">
        <img src={brain} className="icon" alt="Technology Icon"/>
        Technology
      </label>
      <label onClick={changePage} id="entertainment" className="item">
        <img src={movie} className="icon" alt="Entertainment Icon"/>
        Entertainment
      </label>
      <label onClick={changePage} id="science" className="item">
        <img src={rocket} className="icon" alt="Science Icon"/>
        Science
      </label>
      <label onClick={changePage} id="health" className="item">
        <img src={heart} className="icon" alt="Health Icon"/>
        Health
      </label>
    </nav>
  )
}

Menu.propTypes = {
  changePage: PropTypes.func
}

export default Menu;
