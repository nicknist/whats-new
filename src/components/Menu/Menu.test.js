import React from 'react';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('Menu', () => {
  it('should show the menu component as a nav bar', () => {
    const wrapper = shallow(<Menu changePage={jest.fn()}/>)

    expect(wrapper).toMatchSnapshot();
  })
})
