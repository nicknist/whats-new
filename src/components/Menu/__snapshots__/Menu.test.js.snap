// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should show the menu component as a nav bar 1`] = `
<nav
  className="menu"
>
  <label
    className="item"
    id="local"
    onClick={[MockFunction]}
  >
    <img
      alt="Location Icon"
      className="icon"
      src="location.svg"
    />
    Local News
  </label>
  <label
    className="item"
    id="technology"
    onClick={[MockFunction]}
  >
    <img
      alt="Technology Icon"
      className="icon"
      src="brain.png"
    />
    Technology
  </label>
  <label
    className="item"
    id="entertainment"
    onClick={[MockFunction]}
  >
    <img
      alt="Entertainment Icon"
      className="icon"
      src="movie.png"
    />
    Entertainment
  </label>
  <label
    className="item"
    id="science"
    onClick={[MockFunction]}
  >
    <img
      alt="Science Icon"
      className="icon"
      src="rocket.png"
    />
    Science
  </label>
  <label
    className="item"
    id="health"
    onClick={[MockFunction]}
  >
    <img
      alt="Health Icon"
      className="icon"
      src="heart.png"
    />
    Health
  </label>
</nav>
`;
